core/src/tile/tileManager.cpp

```diff
@@ -65,17 +65,22 @@ void TileManager::setTileSources(const std::vector<std::shared_ptr<TileSource>>&
     m_tileSets.erase(it, m_tileSets.end());
 
     // add new sources
-    for (const auto& source : _sources) {
+    for (auto& source : _sources) {
 
-        if (std::find_if(m_tileSets.begin(), m_tileSets.end(),
-                         [&](const TileSet& a) {
-                             return a.source->name() == source->name();
-                         }) == m_tileSets.end()
-                && source->generateGeometry()) {
+        if (source->generateGeometry()) {
 
-            LOGD("add source %s", source->name().c_str());
+            auto sIt = std::find_if(m_tileSets.begin(), m_tileSets.end(),
+                             [&](const TileSet& tileSet) {
+                                        return source->equals(*tileSet.source);
+                             });
 
-            m_tileSets.push_back({ source, false });
+            if (sIt == m_tileSets.end()) {
+                LOGN("Add source %s", source->name().c_str());
+                m_tileSets.push_back({ source, false });
+            } else {
+                LOGN("Reuse source from previous scene %s", source->name().c_str());
+                source = sIt->source;
+            }
```

Comment:
Yeah, we were discussing this for https://github.com/tangrams/tangram-es/pull/1504.